### PR TITLE
Look up the correct top-level resource name when reconstructing foreign keys

### PR DESCRIPTION
**Pull request by jbradberry**
##### SUMMARY

during an awx-cli export.

related #11887 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
- CLI

### CODE REVIEW

File: awxkit/awxkit/api/pages/api.py [hunk 1]
```diff
@@ -100,7 +100,13 @@ def _export(self, _page, post_fields):
             else:
                 if post_fields[key]['type'] == 'id' and _page.json.get(key) is not None:
                     log.warning("Related link %r missing from %s, attempting to reconstruct endpoint.", key, _page.endpoint)
-                    resource = getattr(self, key, None)
+                    res_pattern, resource = getattr(resources, key, None), None
+                    if res_pattern:
+                        try:
+                            top_level = res_pattern.split('/')[3]
+                            resource = getattr(self, top_level, None)
```
Comment (AlanCoding) [inline awxkit/awxkit/api/pages/api.py:107]: This code is kind of hard to read.

The best I can reconstruct, the reason for this is to handle

```python
>>> from awxkit.api.resources import resources
>>> resources.application
'/api/v2/applications/\\d+/'
>>> resources.application.split('/')[3]
'applications'
```

Is that right? This is to add an "s" to "application"?

Comment (jbradberry) [reply awxkit/awxkit/api/pages/api.py:107]: To 'organization', yes.

Comment (jbradberry) [reply awxkit/awxkit/api/pages/api.py:107]: I didn't want to assume that I could just always concatenate an 's' and get the correct endpoint, e.g. 'inventory' -> 'inventories'.

**Review by AlanCoding — APPROVED** [review]

**Comment by jbradberry** [comment]
Closing this for now.

**Pull request merged**